src/Tools/dotnet-monitor/Startup.cs

```diff
@@ -117,6 +118,11 @@ public void ConfigureServices(IServiceCollection services)
                 logger.UnableToListenToAddress(result.Url, result.Exception);
             }
 
+            foreach (var warningMessage in hostBuilderContext.Properties)
+            {
+                logger.LogWarning(warningMessage.Value.ToString());
```

Comment:
I like where this is going. However, I don't think we should be putting values directly in the properties bag since there could be other properties that are added not by our code. I would suggest creating a new object type (e.g. `HostBuilderResults`) that contains a property for warnings (e.g. `public List<string> Warnings { get; } = new()`), add this to `context.Properties` with a well-known key, add whatever issues to the warnings property and then get the object back out here and iteratively log them. This will allow for clear separation of ownership and allow us to expand upon it in the future more easily.

Reply:
I had thought about that issue too, and I think you're right - although, is there a reason why we're creating a new type (such as `HostBuilderResults`) just to hold a list of strings, when presumably couldn't we just have a key (e.g. "Warnings") that maps directly to a list of strings (the warning messages)?

Reply:
As Justin pointed out, it will be easier to add additional data (for example, if we also wanted Errors not just Warnings) in the future if we use an object instead of a List<string>.